### DISCUSSION TITLE
Declarative Automation maximum evaluation per tick

### DISCUSSION THREAD

**Discussion by liongkj** [dagster-io/dagster#24930] [Q&A] [answered]
Hi, how can i set a maximum number of evaluation to do? lets say i would like to have maximum of 50 jobs per automation sensor tick

**Comment by garethbrickman** [comment]
Do you mean you want to limit the number of asset materializations that occur simultaneously when the Automation Condition Sensor evaluates? That should probably be handled with [asset-level concurrency](https://docs.dagster.io/guides/limiting-concurrency-in-data-pipelines#configuring-opasset-level-concurrency). Note that you can [set run_tags as a parameter of the AutomationConditionSensor](https://docs.dagster.io/_apidocs/assets#dagster.AutomationConditionSensorDefinition) to help with that.

**Reply by liongkj** [reply]
No, not sure am i using the correct term. I attached a screenshot. I meant to limit the number of requested materialization. "775" in the below screenshot.
![image](https://github.com/user-attachments/assets/bb0de070-ebe0-4be9-8245-c635e10135ca)

**Reply by garethbrickman** [reply, marked as answer]
The [AutomationConditionSensorDefinition](https://docs.dagster.io/_apidocs/assets#dagster.AutomationConditionSensorDefinition) is going to target a set of assets and repeatedly evaluate all the AutomationConditions on all of those assets. So you would either need to change the number of assets being targeted, or change the AutomationConditions. There is no setting at the level of the AutomationConditionSensor for limiting/queuing the materializations themselves.
